hms-api/src/repository/hackathon-repository.ts

```diff
@@ -32,12 +34,39 @@ export async function listHackathons(): Promise<Hackathon[]> {
 }
 
 export async function putHackathon(hackathon: Hackathon) {
+  if (await hackathonExists(hackathon.id)) {
+    throw new InvalidStateError('Cannot create Hackathon, it already exists');
+  }
+
+  if (await hackathonSlugExists(hackathon.slug)) {
+    throw new InvalidStateError('Cannot create Hackathon, slug already exists');
+  }
+
   await dynamoDBClient.send(
     new PutItemCommand({
       TableName: table,
       Item: {
         title: {S: hackathon.title},
         description: {S: hackathon.description},
+        slug: {S: hackathon.slug},
+        startDate: {S: hackathon.startDate.toISOString()},
+        endDate: {S: hackathon.endDate.toISOString()},
+        id: {S: hackathon.id},
+        creationDate: {S: hackathon.creationDate.toISOString()},
+        votingOpened: {BOOL: hackathon.votingOpened},
+      },
+    }),
+  );
+}
+
+export async function updateHackathon(hackathon: Hackathon) {
```

Comment:
ok I see what you did now, so the way everything was working was that the repository classes are meant to be really dumb.  They just convert a service layer entity (Hackathon, Idea, User, whatever...) into something DynamoDB understands and makes a request to it.  All of the business logic around checking if something exists already or not is done in the service classes

Reply:
my opinion would be to keep it consistent and have the repository classes be dumb and put the smarts in the service layer.  This from a design standpoint has the advantage that if you were say trying to switch from DynamoDB, Redis, or RDS then you don't need to duplicate this business logic in other classes.

This allows for you to have less code overall and be more agile if you wanna change storage options or use multiple for things